api/provider/aws/deploy_read.go

```diff
@@ -0,0 +1,86 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Read(deployID string) (*models.Deploy, error) {
+	deployModel, err := d.newDeployModel(deployID)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefinitionARN, err := d.lookupTaskDefinitionARN(deployID)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefinitionOutput, err := d.describeTaskDefinition(taskDefinitionARN)
+	if err != nil {
+		return nil, err
+	}
+
+	deployFile, err := json.Marshal(taskDefinitionOutput)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to extract deploy file: %s", err.Error())
+	}
+
+	deployModel.DeployFile = deployFile
+
+	return deployModel, nil
+}
+
+func (d *DeployProvider) describeTaskDefinition(taskDefinitionARN string) (*ecs.TaskDefinition, error) {
+	input := &ecs.DescribeTaskDefinitionInput{}
+	input.SetTaskDefinition(taskDefinitionARN)
+
+	output, err := d.AWS.ECS.DescribeTaskDefinition(input)
```

Comment:
this should catch when the task does not exist

Reply:
:white_check_mark: 

I went with

```     output, err := d.AWS.ECS.DescribeTaskDefinition(input)
     if err != nil {
         if err, ok := err.(awserr.Error); ok && strings.Contains(err.Message(), "Unable to describe task definition") {
             return nil, errors.Newf(errors.DeployDoesNotExist, "Deploy '%s' does not exist", taskDefinitionARN)
         }
 
         return nil, err
     }```